src/query/executor/engine.go

```diff
@@ -132,7 +133,7 @@ func (e *Engine) ExecuteExpr(ctx context.Context, parser parser.Parser, opts *En
 	}()
 
 	req := newRequest(e, params)
-	defer req.finish()
+
```

Comment:
Should this still be deferring the finish?

Reply:
Maybe add a quick comment saying that downstream methods will be closing the request?

Also what happens if we hit an error on `req.compile` or `req.plan`, will we miss the finish call?